compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -197,16 +197,14 @@ object Inliner {
     /** Expand call to scala.compiletime.testing.typeChecks */
     def typeChecks(tree: Tree)(implicit ctx: Context): Tree = {
       assert(tree.symbol == defn.CompiletimeTesting_typeChecks)
-      def getCodeArgValue(t: Tree): Option[String] = t match {
-        case Literal(Constant(code: String)) => Some(code)
-        case Typed(t2, _) => getCodeArgValue(t2)
-        case Inlined(_, Nil, t2) => getCodeArgValue(t2)
-        case Block(Nil, t2) => getCodeArgValue(t2)
-        case _ => None
+      def stripTyped(t: Tree): Tree = t match {
+        case Typed(t2, _) => stripTyped(t2)
+        case _ => t
       }
+
       val Apply(_, codeArg :: Nil) = tree
-      getCodeArgValue(codeArg.underlyingArgument) match {
-        case Some(code) =>
+      ConstFold(stripTyped(codeArg.underlyingArgument)).tpe.widenTermRefExpr match {
```

Comment:
If we call `ConstFold` here, should we add a test like `typeChecks("1" + "1")`?

Reply:
Added it to `tests/run/i7487.scala`